hw/ip/flash_ctrl/doc/_index.md

```diff
@@ -65,40 +75,130 @@ The physical controller supports the following features
+Instead it is organized into data partitions and information partitions.
+
+The data partition holds generic data like a generic memory would.
+The information partition typically holds metadata about the data partition.
+This includes but is not limited to information such as
+*  Redundancy information.
+*  Manufacturer specific information.
+*  Manufacturer flash timing information.
+
+Note, there **can** be more than one information partition, and none of them are required to be the same size as the data partition.
+See the diagram below for an illustrative example.
+![Flash Example Parittion](flash_partitions.svg)
+
+Further, while the different partitions may be identical in some attributes, they are different in others.
+*  All types of partitions must have the same page size and word size.
+*  All types of partitions obey the same program and erase rules :
+   * A bit cannot be programmed back to 1 once it has been programmed to 0.
+   * Only erase can restore a bit to 1 under normal circumstances.
+*  Date partitions can be directly read by software and other hardware hosts, while information partitions can only be read by the flash controller
+
+By default, this design assumes 1 type of information partition and 4 pages per type of information partition.
 
 
 # Theory of Operation
 
 ## Block Diagram
 
+![Flash Block Diagram](flash_block_diagram.svg)
+
 ### Flash Protocol Controller
 
-The Flash Protocol Controller sits between the host software interface and the flash physical controller, which contains the physical flash.
-Its primary function is to translate software requests into a high level protocol for the actual flash block.
-Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash.
-Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls.
+The Flash Protocol Controller sits between the host software interface, other hardware components and the flash physical controller.
+Its primary functions are two fold
+*  Translate software program, erase and read requests into a high level protocol for the actual flash physical controller
+*  Act as communication interface between flash and other components in the system, such as life cycle and key manager.
+
+Importantly, the flash protocol controller is not responsible for the detailed timing and waveform control of the flash, nor is it responsible for data scrambling and reliability metadata such as parity and ECC.
+Instead, it maintains FIFOs / interrupts for the software to process data, as well as high level abstraction of region protection controls as well as error handling.
+
+There are two hardware connections between the flash controller and the rest of the design.
+*  The first is to life cycle, in which flash content plays a part.
+*  The second is to key manager, whose seed material may be partially sourced from flash.
```

Comment:
(Apologies if this is answered somewhere) How does key manager retrieve this info?